### DISCUSSION TITLE
Unable to start container

### DISCUSSION THREAD

**Discussion by marvic2409** [haugene/docker-transmission-openvpn#2202] [Q&A] [answered]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [X] I have tried using dev branch

### Docker run config used

docker run --cap-add=NET_ADMIN -d               -v /nvme/torrent/config/openvpn/AirVPN_United-States_UDP-443.ovpn:/etc/openvpn/custom/default.ovpn -v /nvme/torrent/data:/data               -e OPENVPN_PROVIDER=CUSTOM               -e LOCAL_NETWORK=192.168.1.0/24           -e DEBUG=true              -p 9091:9091           haugene/transmission-openvpn

### Current Behavior

When I start the container, it tries to start for 20 ish seconds and then dies.

### Expected Behavior

To start properly

### How have you tried to solve the problem?

Added --dns

### Log output

Starting container with revision: d1ece1c5f36f275d5d950bd2557ec43f79a80d14
+ [[ -x /scripts/openvpn-pre-start.sh ]]
+ compgen -e
+ grep -q OVERRIDE_DNS
+ nslookup google.com
+ echo 'WARNING: initial DNS resolution test failed'
+ [[ true == \t\r\u\e ]]
+ echo 'Creating TUN device /dev/net/tun'
+ mkdir -p /dev/net
WARNING: initial DNS resolution test failed
Creating TUN device /dev/net/tun
+ mknod /dev/net/tun c 10 200
+ chmod 0666 /dev/net/tun
+ VPN_PROVIDER=CUSTOM
+ export VPN_PROVIDER=custom
+ VPN_PROVIDER=custom
+ export VPN_PROVIDER_HOME=/etc/openvpn/custom
+ VPN_PROVIDER_HOME=/etc/openvpn/custom
+ mkdir -p /etc/openvpn/custom
+ [[ -z '' ]]
+ [[ CUSTOM == \*\*\N\o\n\e\*\* ]]
+ [[ -z CUSTOM ]]
Using OpenVPN provider: CUSTOM
+ echo 'Using OpenVPN provider: CUSTOM'
+ [[ -n '' ]]
+ [[ -z '' ]]
+ VPN_CONFIG_SOURCE=auto
+ VPN_CONFIG_SOURCE=auto
+ echo 'Running with VPN_CONFIG_SOURCE auto'
+ [[ auto == \a\u\t\o ]]
+ [[ -x /etc/openvpn/custom/configure-openvpn.sh ]]
Running with VPN_CONFIG_SOURCE auto
+ echo 'No bundled config script found for CUSTOM. Defaulting to external config'
+ VPN_CONFIG_SOURCE=external
+ [[ external == \e\x\t\e\r\n\a\l ]]
+ ./etc/openvpn/fetch-external-configs.sh
No bundled config script found for CUSTOM. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.U89BmZ8FI2
curl: (6) Could not resolve host: github.com
./etc/openvpn/fetch-external-configs.sh: line 16: config_repo_temp_dir: unbound variable
+ [[ -x /etc/openvpn/custom/configure-openvpn.sh ]]
+ [[ -z '' ]]
+ [[ -n '' ]]
No VPN configuration provided. Using default.
+ echo 'No VPN configuration provided. Using default.'
+ CHOSEN_OPENVPN_CONFIG=/etc/openvpn/custom/default.ovpn
+ MODIFY_CHOSEN_CONFIG=true
+ [[ true == \t\r\u\e ]]
+ /etc/openvpn/modify-openvpn-config.sh /etc/openvpn/custom/default.ovpn
Modifying /etc/openvpn/custom/default.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
sed: cannot rename /etc/openvpn/custom/sedQ4TCCY: Device or resource busy
Modification: Change ca certificate path
sed: cannot rename /etc/openvpn/custom/sedRIcDb4: Device or resource busy
Modification: Change ping options
sed: cannot rename /etc/openvpn/custom/sediWC7Wa: Device or resource busy
sed: cannot rename /etc/openvpn/custom/sedKfUF5e: Device or resource busy
sed: cannot rename /etc/openvpn/custom/sedphmPVk: Device or resource busy
sed: cannot rename /etc/openvpn/custom/sed2GbR9l: Device or resource busy
Modification: Update/set resolv-retry to 15 seconds
sed: cannot rename /etc/openvpn/custom/sed6lAAMu: Device or resource busy
sed: cannot rename /etc/openvpn/custom/sedOJDMiw: Device or resource busy
Modification: Change tls-crypt keyfile path
sed: cannot rename /etc/openvpn/custom/sed2pxA5F: Device or resource busy
Modification: Set output verbosity to 3
sed: cannot rename /etc/openvpn/custom/sediXrVzN: Device or resource busy
sed: cannot rename /etc/openvpn/custom/sedpuDY9P: Device or resource busy
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
sed: cannot rename /etc/openvpn/custom/sedXm5yzU: Device or resource busy
sed: cannot rename /etc/openvpn/custom/sedqcwQZX: Device or resource busy
+ [[ -x /scripts/openvpn-post-config.sh ]]
+ [[ '' == \*\*\N\o\n\e\*\* ]]
+ [[ '' == \*\*\N\o\n\e\*\* ]]
Setting OpenVPN credentials...
+ echo 'Setting OpenVPN credentials...'
+ mkdir -p /config
+ echo ''
+ echo ''
+ chmod 600 /config/openvpn-credentials.txt
+ echo ''
+ echo ''
+ python3 /etc/openvpn/persistEnvironment.py /etc/transmission/environment-variables.sh
+ TRANSMISSION_CONTROL_OPTS='--script-security 2 --up-delay --up /etc/openvpn/tunnelUp.sh --route-pre-down /etc/openvpn/tunnelDown.sh'
+ [[ false == \t\r\u\e ]]
+ [[ -n 192.168.1.0/24 ]]
++ /sbin/ip route list match 0.0.0.0
++ awk '{if($5!="tun0"){print "GW="$3"\nINT="$5; exit}}'
+ eval GW=172.17.0.1 INT=eth0
++ GW=172.17.0.1
++ INT=eth0
+ [[ false == \t\r\u\e ]]
+ [[ false == \t\r\u\e ]]
+ [[ -n 192.168.1.0/24 ]]
+ [[ -n 172.17.0.1 ]]
+ [[ -n eth0 ]]
+ for localNet in ${LOCAL_NETWORK//,/ }
+ echo 'adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0'
adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0
+ /sbin/ip route add 192.168.1.0/24 via 172.17.0.1 dev eth0
+ [[ false == \t\r\u\e ]]
+ [[ false != \f\a\l\s\e ]]
+ exec openvpn --script-security 2 --up-delay --up /etc/openvpn/tunnelUp.sh --route-pre-down /etc/openvpn/tunnelDown.sh --config /etc/openvpn/custom/default.ovpn
Wed Mar  2 15:07:51 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021
Wed Mar  2 15:07:51 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Wed Mar  2 15:07:51 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Wed Mar  2 15:07:51 2022 Outgoing Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
Wed Mar  2 15:07:51 2022 Incoming Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
Wed Mar  2 15:08:03 2022 RESOLVE: Cannot resolve host address: us.vpn.airdns.org:443 (Temporary failure in name resolution)
Wed Mar  2 15:08:15 2022 RESOLVE: Cannot resolve host address: us.vpn.airdns.org:443 (Temporary failure in name resolution)
Wed Mar  2 15:08:33 2022 RESOLVE: Cannot resolve host address: us.vpn.airdns.org:443 (Temporary failure in name resolution)
Wed Mar  2 15:08:50 2022 RESOLVE: Cannot resolve host address: us.vpn.airdns.org:443 (Temporary failure in name resolution)
Wed Mar  2 15:08:50 2022 Could not determine IPv4/IPv6 protocol
Wed Mar  2 15:08:50 2022 SIGTERM[soft,init_instance] received, process exiting

### HW/SW Environment

```markdown
- OS: Arch Linux
- Docker:20.10.12
```

### Anything else?

I am using AirVPN with a valid configuration file

**Comment by pkishino** [comment, marked as answer]
This is not an issue but a config question..Please don't open issues for support requests.. that is what the Q&A is for.
Please look at your log, your container cannot resolve the DNS query. 
Check your docker setup, try setting specific DNS, check the documentation for troubleshooting

**Reply by marvic2409** [reply]
I tried setting a specific DNS and it still gave the same error

**Reply by pkishino** [reply]
well, the initial dns lookup is failing. that lookup goes directly out based on whatever dns your network uses etc or that you specify. If this one fails then you need to check your network setup etc.
You could try to deploy a basic hello world container and see if nslookup fails as well.

**Reply by marvic2409** [reply]
Ah sorry, I was going to delete that reply, turns out restarting the docker service fixed the issue. Looks like it was an issue with the current docker service.
